src/SOS/Strike/util.cpp

```diff
@@ -3225,21 +3277,64 @@ size_t FunctionType (size_t EIP)
     return (size_t) pMD;
 }
 
-#ifndef FEATURE_PAL
-
 //
 // Gets version info for the CLR in the debuggee process.
 //
-BOOL GetEEVersion(VS_FIXEDFILEINFO *pFileInfo)
+BOOL GetEEVersion(VS_FIXEDFILEINFO* pFileInfo, char* fileVersionBuffer, int fileVersionBufferSize)
 {
-    _ASSERTE(g_ExtSymbols2);
     _ASSERTE(pFileInfo);
+    if (g_ExtSymbols2 == nullptr) {
+        return FALSE;
+    }
+    ModuleInfo moduleInfo = g_moduleInfo[GetEEFlavor()];
+    _ASSERTE(moduleInfo.index != DEBUG_ANY_ID);
+
+#ifdef FEATURE_PAL
+    // Load the symbols for runtime. On Linux we are looking for the "sccsid" 
+    // global so "libcoreclr.so/.dylib" symbols need to be loaded.
+    LoadNativeSymbols(true);
+#endif
 
-    // Grab the version info directly from the module.
-    return g_ExtSymbols2->GetModuleVersionInformation(
-        DEBUG_ANY_ID, g_moduleInfo[GetEEFlavor()].baseAddr, "\\", pFileInfo, sizeof(VS_FIXEDFILEINFO), NULL) == S_OK;
+    HRESULT hr = g_ExtSymbols2->GetModuleVersionInformation(moduleInfo.index, 0, "\\", pFileInfo, sizeof(VS_FIXEDFILEINFO), NULL);
+
+    // Attempt to get the the FileVersion string that contains version and the "built by" and commit id info
+    if (fileVersionBuffer != nullptr)
+    {
+        if (fileVersionBufferSize > 0) {
+            fileVersionBuffer[0] = '\0';
+        }
+        // We can assume the English/CP_UNICODE lang/code page for the runtime modules
+        g_ExtSymbols2->GetModuleVersionInformation(
+            moduleInfo.index, 0, "\\StringFileInfo\\040904B0\\FileVersion", fileVersionBuffer, fileVersionBufferSize, NULL);
+    }
+
+    return SUCCEEDED(hr);
+}
+
+//
+// Return true if major runtime version
```

Comment:
Maybe comment what major version this is referring to? The naïve guess would be it refers to major version in the version number of the binary, but I assume what you meant here is the logical major version we used in our product marketing.

Reply:
Added comment.